### ISSUE TITLE
Default snapshotting behaviour breaks RStudio's `devtools::load_all()` keyboard shortcut

### ISSUE THREAD

**Issue by nteetor** [rstudio/renv#1742]
Within a package development environment, {renv} does not typically snapshot a package like {devtools}. Unfortunately, this seems to break RStudio's `devtools::load_all()` keyboard shortcut (i.e. nothing happens). I was able to fix the problem with the following steps,

1. Configure renv to capture all packages in the project library, `renv::settings$snapshot.type("all")`
2. Snapshot packages, `renv::snapshot()`
3. Restart R session

A month or so ago I opened a related issue under the rstudio/rstudio repo, https://github.com/rstudio/rstudio/issues/13727.

**Comment by kevinushey** [comment]
As a workaround, you can add a line like `library(devtools)` to a file called `_dependencies.R` in the project. (You may need to include it in `.Rbuildignore` to make sure it doesn't get built with the package.)

**Comment by nteetor** [comment]
True, thank you for pointing that out. I'd resigned myself to always using `renv::settings$snapshot.type("all")`.

**Comment by kevinushey** [comment]
I think the workaround above is sufficient for this particular issue, so I'm going to close -- please let me know if you're still having trouble.
